ani-cli

```diff
@@ -1,656 +1,238 @@
-	[ "$debug" -eq 1 ] && exit 0
-	# write anime and episode number and save to temporary history
-	grep -q -- "$selection_id" "$logfile" || printf "%s\t%s\n" "$selection_id" $((episode+1)) >> "$logfile"
-	sed -E "s/^${selection_id}\t[0-9]*/${selection_id}\t$((episode+1))/" "$logfile" > "${logfile}.new"
-	[ ! "$PID" = "0" ] && kill "$PID" >/dev/null 2>&1
-	[ -z "$video_url" ] && die "Video URL not found"
-	trackma_title="$(printf '%s' "$anime_id Episode $episode" | tr '-' ' ' | awk '{for(i=1;i<=NF;i++){ $i=toupper(substr($i,1,1)) substr($i,2) }}1')"
-	if [ "$auto_play" -eq 0 ]; then
-		play_episode "$video_url" "$refr" "$trackma_title"
-	else
-		printf "\n"
-		play_episode "$video_url" "$refr" "$trackma_title"
-		wait
-		sleep 2
-	fi
-	PID=$!
-	# overwrite history with temporary history
-	mv "${logfile}.new" "$logfile"
+download () {
+	case $1 in
+		*m3u8*)
+			ffmpeg -loglevel error -stats -i "$1" -c copy "$download_dir/$2.mp4" ;;
+		*)
+			aria2c --check-certificate=false --summary-interval=0 -x 16 -s 16 "$1" --dir="$download_dir" -o "$2.mp4" --download-result=hide ;;
+	esac
 }
 
-play_episode () {
-	video_url="$1"; refr="$2" trackma_title="$3"
-	if [ "$player_fn" = "download" ];then
-		inf "Currently downloading $trackma_title ($provider_name)"
-	else
-		inf "Currently playing $trackma_title ($provider_name)"
-	fi
-	case "$player_fn" in
-		download)
-			if download "$refr" "$video_url" "$trackma_title"; then
-				inf "Downloaded episode: $trackma_title"
-			else
-				err "Download failed episode: $trackma_title , please retry or check your internet connection"
-			fi
-			;;
-		iina)
-			nohup "$player_fn" "$video_url" --no-stdin --keep-running --mpv-referrer="$refr" --mpv-force-media-title="$trackma_title" > /dev/null 2>&1 & ;;
-		vlc)
-			if uname -a | grep -qE '[Aa]ndroid';then
-				am start --user 0 -a android.intent.action.VIEW -d "$video_url" -n org.videolan.vlc/org.videolan.vlc.gui.video.VideoPlayerActivity -e "title" "$trackma_title" > /dev/null 2>&1 &
-			else
-				nohup "$player_fn" "$video_url"  --http-referrer="$refr" --meta-title "$trackma_title" --play-and-exit > /dev/null 2>&1 &
-			fi
-			;;
-		"syncplay"|"/Applications/Syncplay.app/Contents/MacOS/syncplay"|"/c/Program Files (x86)/Syncplay/Syncplay.exe")
-			nohup "$player_fn" "$video_url" -- --referrer="$refr" --force-media-title="$trackma_title" > /dev/null 2>&1 & ;;
-		*)
-			if uname -a | grep -qE '[Aa]ndroid';then
-				am start --user 0 -a android.intent.action.VIEW -d "$video_url" -n is.xyz.mpv/.MPVActivity > /dev/null 2>&1 &
-			else
-				nohup "$player_fn" "$video_url" --referrer="$refr" --force-media-title="$trackma_title" > /dev/null 2>&1 &
-			fi
-			;;
+play_episode() {
+	get_episode_url
+	#play the episode
+	case "$player_function" in
+	debug)
+		printf "All links:\n"
+		cat "$cache_dir"/*
+		printf "Selected link:\n%s\n" "$episode"
+		;;
+	mpv | iina)
+		nohup "$player_function" --force-media-title="${allanime_title}episode-${ep_no}-${mode}" "$episode" > /dev/null &
```

Comment:
consider removing the "&" at the end
I don't see a reason the menu should be shown when something is already playing and you can't interact with the player when rofi is open if you're using external menu

Reply:
Detaching from the main process is useful not only because this is the behaviour everyone got used to but this way the script's process can exit, and the player will stay open. I don't see a reason why we shouldn't.

Reply:
Then I saw how rofi works and I totally take back what I said. [This commit](https://github.com/pystardust/ani-cli/pull/955/commits/84001a548423540e291ef3350abff96a123c8c95) should address the matter